src/DataTable/Checkbox.js

```diff
@@ -20,7 +21,7 @@ const Checkbox = ({
   };
 
   const TagName = component;
-  const baseStyle = TagName !== 'input' ? componentOptions.style : baseCheckboxStyle;
+  const baseStyle = TagName !== 'input' ? componentOptions.style : { ...baseCheckboxStyle, cursor: disabled ? 'not-allowed' : 'pointer' };
```

Comment:
hmm, this will cause `Checkbox` to re-render since the style Object is being recreated on every render pass.

Reply:
I don't know how to avoid that and don't much have idea on `styled-components` but because of we have dynamic component styling might be recomputed anyway

let me know if you have any solution.

Reply:
You would memoize - in this case with the `useMemo` hook to cache the calculation. However, on second thought just leave it as it is and I'll refactor in a separate PR.

Reply:
Additionally, I need to check if there is a performance benefit to memoizing